pkg/key/key.go

```diff
@@ -1,22 +1,109 @@
 package key
 
 import (
+	"context"
 	"fmt"
+	"io/ioutil"
+	"os"
 
-	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	chartv1 "github.com/giantswarm/apiextensions-application/api/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	clientgoscheme "k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/tools/clientcmd"
+	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
 const (
-	ClusterNameLabel        = "cluster.x-k8s.io/cluster-name"
-	ClusterWatchFilterLabel = "cluster.x-k8s.io/watch-filter"
+	ClusterNameLabel = "cluster.x-k8s.io/cluster-name"
 
 	FinalizerName = "encryption-provider-operator.finalizers.giantswarm.io"
+
+	MasterNodeLabel = "node-role.kubernetes.io/master"
 )
 
-func GetClusterIDFromLabels(t v1.ObjectMeta) string {
+func GetClusterIDFromLabels(t metav1.ObjectMeta) string {
 	return t.GetLabels()[ClusterNameLabel]
 }
 
 func SecretName(clusterName string) string {
 	return fmt.Sprintf("%s-encryption-provider-config", clusterName)
 }
+
+// GetWCK8sClient will return workload cluster k8s controller-runtime client
+func GetWCK8sClient(ctx context.Context, ctrlClient client.Client, clusterName string, clusterNamespace string) (client.Client, error) {
+	var err error
+
+	if _, err := os.Stat(tempKubeconfigFileName(clusterName)); err == nil {
```

Comment:
can the kubeconfig expire for some reason?

Reply:
Well theoretically yes, the certs used for kubeconfig has an expiration date, for our product, it's 180 days for capi it is less but still, its very unlikely the lifetime of the pod will be longer than the certificate expiration date

I just copied this part from another component,  trying to void that the kubeconfig has to be created every time, but maybe it's not so bad idea to just generate a new one every time, would that make you more happy?

Reply:
I am happy in any case :D your call really